File: SEGUNDO/PDOO/PRACTICAS/PRACTICA2/java/deepSpace/src/deepspace/PowerEfficientSpaceStation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author lauralazaro
 */
public class PowerEfficientSpaceStation extends SpaceStation {
    private final float EFFICIENCYFACTOR =1.10f;
    
    PowerEfficientSpaceStation(SpaceStation station) {
        super(station);
    }
    
    @Override public float fire() {
        return super.fire()*EFFICIENCYFACTOR;
    }
    
    @Override public float protection() {
        return super.protection()*EFFICIENCYFACTOR;
    }
    
    @Override public Transformation setLoot(Loot loot) {
        Transformation check = super.setLoot(loot);
        
        if (check==Transformation.SPACECITY)
            return Transformation.NOTRANSFORM;
        else
            return check;
    }
    
}

File: SEGUNDO/PDOO/PRACTICAS/practicar/ruby/PowerEfficientSpaceStation.rb
#encoding:utf-8

require 'singleton'

require_relative 'SpaceStation.rb'
require_relative 'Transformation.rb'

module Deepspace
    class PowerEfficientSpaceStation < SpaceStation
        @@EFFICIENCYFACTOR=1.1
        def initialize (station)
            newCopy(station)
        end

        def fire
            fire=super
            return (fire*EFFICIENCYFACTOR)
        end

        def protection
            shield = super
            return (shield*EFFICIENCYFACTOR)
        end

        def setLoot(loot)
            check = super

            if (check==Transformation::SPACECITY)
                return Transformation::NOTRANSFORM
            else
                return check
            end
        end
    end
end